docs/index.qmd

```diff
@@ -0,0 +1,234 @@
+2.  **Reproducibility**: Encapsulated Python dependencies for each workflow
+3.  **Parameter Sweeping**: Matrix execution patterns for systematic exploration across tasks and models
+
+Inspect Flow is designed for researchers and engineers running systematic AI evaluations who need to scale beyond ad-hoc scripts.
+
+## Getting Started
+
+::: callout-note
+### Prerequisites
+
+Before using Inspect Flow, you should:
+
+-   Have familiarity with [Inspect AI](https://inspect.aisi.org.uk/)
+-   Have an existing Inspect evaluation or use one from [inspect-evals](https://github.com/UKGovernmentBEIS/inspect_evals)
+:::
+
+### Installation
+
+Install the `inspect_flow` package from GitHub as follows:
+
+``` bash
+pip install git+https://github.com/meridianlabs-ai/inspect_flow
+```
+
+### Set up API keys
+
+You'll need API keys for the model providers you want to use. Set the relevant provider API key in your `.env` file or export it in your shell:
+
+::: {.panel-tabset .code-tabset}
+#### OpenAI
+
+``` bash
+export OPENAI_API_KEY=your-openai-api-key
+```
+
+#### Anthropic
+
+``` bash
+export ANTHROPIC_API_KEY=your-anthropic-api-key
+```
+
+#### Google
+
+``` bash
+export GOOGLE_API_KEY=your-google-api-key
+```
+
+#### Grok
+
+``` bash
+export GROK_API_KEY=your-grok-api-key
+```
+
+#### Mistral
+
+``` bash
+export MISTRAL_API_KEY=your-mistral-api-key
+```
+
+#### Hugging Face
+
+``` bash
+export HF_TOKEN=your-hf-token
+```
+:::
+
+### Optional: VS Code extension
+
+Optionally install the [Inspect AI VS Code Extension](https://inspect.aisi.org.uk/vscode.html) which includes features for viewing evaluation log files.
+
+## Basic Examples
+
+Let's walk through creating your first Flow configuration. We'll use `FlowConfig` (the entrypoint class) and `FlowTask` to define evaluations.
+
+::: {.callout-tip collapse="true"}
+### Core Components Reference
+
+-   `types.FlowConfig` — Pydantic class that encapsulates the declarative description of a Flow job.
+-   `types.FlowTask` — Pydantic class abstraction on top of Inspect AI [Task](https://inspect.aisi.org.uk/tasks.html).
+-   `types.FlowModel` — Pydantic class abstraction on top of Inspect AI [Model](https://inspect.aisi.org.uk/models.html).
+-   `types.FlowGenerateConfig` — Pydantic class abstraction on top of Inspect AI [GenerateConfig](https://inspect.aisi.org.uk/models.html#generation-config).
+-   `tasks_matrix` — Helper function for parameter sweeping to generate a list of tasks with all parameter combinations.
+-   `models_matrix` — Helper function for parameter sweeping to generate a list of models with all parameter combinations.
+-   `configs_matrix` — Helper function for parameter sweeping to generate a list of GenerateConfig with all parameter combinations.
+:::
+
+`FlowConfig` is the main entrypoint for defining evaluation runs. At its core, it takes a list of tasks to run. Here's a simple example that runs two evaluations:
+
+``` {.python filename="config.py"}
```

Comment:
I think all files should reference and include inline files in our examples dir. That will be simpler to maintain.